dask/array/core.py

```diff
@@ -1497,6 +1497,16 @@ def __xor__(self, other):
     def __rxor__(self, other):
         return elemwise(operator.xor, other, self)
 
+    def __matmul__(self, other):
+        return tensordot(self, other, axes=((self.ndim - 1,),
+                                            (other.ndim - 2,)))
+
+    def __rmatmul__(self, other):
+        if not hasattr(other, 'ndim'):
+            other = np.asarray(other)  # account for array-like on LHS
+        return tensordot(other, self, axes=((other.ndim - 1,),
+                                            (self.ndim - 2,)))
```

Comment:
Is this choice to convert if other doesn't have an `.ndim` attribute standard?

This question also came up on https://github.com/mrocklin/sparse/pull/14 .  It might be interesting to look at the discussion there.  Honestly I have forgotten most of the details of that conversation.